src/Raven.Server/Documents/Handlers/PullReplicationHandler.cs

```diff
@@ -57,33 +57,43 @@ public async Task RegisterHubAccess()
 
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             {
-                var blittableJson = await context.ReadForMemoryAsync(RequestBodyStream(), "register-hub-access");
-                var access = JsonDeserializationClient.ReplicationHubAccess(blittableJson);
-                access.Validate();
-
-                var definition = Database.GetPullReplicationDefinition(hub);
-                if (definition == null)
+                var hubDefinition = Database.GetPullReplicationDefinition(hubTaskName);
+                if (hubDefinition == null)
                 {
                     HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
                     return;
                 }
 
 #pragma warning disable CS0618 // Type or member is obsolete
-                if (definition.Certificates != null && definition.Certificates.Count > 0)
+                if (hubDefinition.Certificates != null && hubDefinition.Certificates.Count > 0)
 #pragma warning restore CS0618 // Type or member is obsolete
                 {
-                    // this handles the backward compact aspect
-                    throw new InvalidOperationException("Cannot register hub access to a replication hub that already has inline certificates: " + hub +
+                    // handle backward compatibility
+                    throw new InvalidOperationException("Cannot register hub access to a replication hub that already has inline certificates: " + hubTaskName +
                                                         ". Create a new replication hub and try again");
                 }
+                
+                var blittableJson = await context.ReadForMemoryAsync(RequestBodyStream(), "register-hub-access");
+                var access = JsonDeserializationClient.ReplicationHubAccess(blittableJson);
+                access.Validate(hubDefinition.WithFiltering);
 
                 using var cert = new X509Certificate2(Convert.FromBase64String(access.CertificateBase64));
                 var publicKeyPinningHash = cert.GetPublicKeyPinningHash();
 
-                var command = new RegisterReplicationHubAccessCommand(Database.Name, hub, access, publicKeyPinningHash, cert.Thumbprint, GetRaftRequestIdFromQuery(),
+                var command = new RegisterReplicationHubAccessCommand(Database.Name, hubTaskName, access, publicKeyPinningHash, cert.Thumbprint, GetRaftRequestIdFromQuery(),
                     cert.Issuer, cert.Subject, cert.NotBefore, cert.NotAfter);
+                
                 var result = await Server.ServerStore.SendToLeaderAsync(command);
                 await WaitForIndexToBeApplied(context, result.Index);
+
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
```

Comment:
FYI, OK is a default status code. You do not need to set it.

Reply:
done